{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import io\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/TYR/table_for_paper.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-906eec50d759>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mMC1R\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/MC1R/table_for_paper.csv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mTYR\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/TYR/table_for_paper.csv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mMTAP\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/MTAP/table_for_paper.csv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mCASP8\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/CASP8/table_for_paper.csv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mOCA2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/OCA2/table_for_paper.csv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'POS'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/Genoma/drobles/ccastaneda/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    684\u001b[0m     )\n\u001b[1;32m    685\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 686\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    687\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    688\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/Genoma/drobles/ccastaneda/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    450\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    451\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 452\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfp_or_buf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    454\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/Genoma/drobles/ccastaneda/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    934\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    935\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 936\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    937\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    938\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/Genoma/drobles/ccastaneda/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1166\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"c\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1167\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"c\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1168\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1169\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1170\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"python\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/mnt/Genoma/drobles/ccastaneda/anaconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1996\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"usecols\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1997\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1998\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1999\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2000\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/TYR/table_for_paper.csv'"
     ]
    }
   ],
   "source": [
    "MC1R = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/MC1R/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')\n",
    "TYR = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/TYR/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')\n",
    "MTAP = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/MTAP/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')\n",
    "CASP8 = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/CASP8/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')\n",
    "OCA2 = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/OCA2/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')\n",
    "CCND1 = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/CCND1/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')\n",
    "CDH1 = pd.read_csv('/mnt/Genoma/drobles/ccastaneda/gwas-reseq/lcr_filtered/regions/CDH1/table_for_paper.csv', header = None).rename(columns = {0:'POS'}).set_index('POS')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "region = str(input(\"enter the region you want to make a table for:\"))\n",
    "wd = '/mnt/Genoma/drobles/ccastaneda/gwas-reseq/regions/'+region\n",
    "os.chdir(wd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "r1 = pd.read_csv('r1').drop_duplicates('Existing_variation')\n",
    "r2 = pd.read_csv('r2').drop_duplicates('Existing_variation')\n",
    "leads = pd.read_csv('to_condition', header = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "r1['38POS'] = r1.GRCh38.str.split('-').str.get(0)\n",
    "r2['38POS'] = r2.GRCh38.str.split('-').str.get(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "leads = leads.rename(columns = {0:'38POS'})\n",
    "leads['GRCh38'] = leads['38POS'] + '-' + leads['38POS'].str.split(':').str.get(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "r1_positions = r1['GRCh38']\n",
    "r2_positions = r2['GRCh38']\n",
    "leads_positions = leads['GRCh38']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.concat([r1_positions, r2_positions, leads_positions], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.to_frame().reset_index().drop(axis = 1, labels = 'index')\n",
    "df.to_csv('haplotypes_38.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "s = r1.merge(r2, how = 'outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Signal</th>\n",
       "      <th>Region</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>GRCh38</th>\n",
       "      <th>ref_alt</th>\n",
       "      <th>-logP_fisher</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Feature</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>OR_fisher</th>\n",
       "      <th>OR_cond1</th>\n",
       "      <th>P_fisher</th>\n",
       "      <th>P_cond1</th>\n",
       "      <th>38POS</th>\n",
       "      <th>ref_alt_x</th>\n",
       "      <th>OR_cond2</th>\n",
       "      <th>P_cond2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>s1</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs1108064</td>\n",
       "      <td>chr16:89827742-89827742</td>\n",
       "      <td>C_G</td>\n",
       "      <td>3.561616</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000378247</td>\n",
       "      <td>YES</td>\n",
       "      <td>1.253</td>\n",
       "      <td>1.108</td>\n",
       "      <td>2.744000e-04</td>\n",
       "      <td>0.108300</td>\n",
       "      <td>chr16:89827742</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>s1</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs1805007&amp;CM981238</td>\n",
       "      <td>chr16:89919709-89919709</td>\n",
       "      <td>T_C</td>\n",
       "      <td>11.147276</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000315491</td>\n",
       "      <td>YES</td>\n",
       "      <td>1.899</td>\n",
       "      <td>1.000</td>\n",
       "      <td>7.124000e-12</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>chr16:89919709</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>s1</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs74800773</td>\n",
       "      <td>chr16:89958562-89958562</td>\n",
       "      <td>G_C</td>\n",
       "      <td>9.714218</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000564836</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.815</td>\n",
       "      <td>1.226</td>\n",
       "      <td>1.931000e-10</td>\n",
       "      <td>0.250100</td>\n",
       "      <td>chr16:89958562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>s1</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs45610233</td>\n",
       "      <td>chr16:89981987-89981987</td>\n",
       "      <td>T_C</td>\n",
       "      <td>9.341321</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000423742</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.845</td>\n",
       "      <td>1.063</td>\n",
       "      <td>4.557000e-10</td>\n",
       "      <td>0.791600</td>\n",
       "      <td>chr16:89981987</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s1</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs73283859</td>\n",
       "      <td>chr16:89996112-89996112</td>\n",
       "      <td>C_G</td>\n",
       "      <td>10.352324</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000437774</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.885</td>\n",
       "      <td>1.259</td>\n",
       "      <td>4.443000e-11</td>\n",
       "      <td>0.328200</td>\n",
       "      <td>chr16:89996112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>s2</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs1800282&amp;COSV66881217</td>\n",
       "      <td>chr16:89816599-89816599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.776504</td>\n",
       "      <td>missense_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>6.0</td>\n",
       "      <td>V/D</td>\n",
       "      <td>ENST00000563513</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.364</td>\n",
       "      <td>1.471</td>\n",
       "      <td>1.673000e-03</td>\n",
       "      <td>0.000107</td>\n",
       "      <td>chr16:89816599</td>\n",
       "      <td>T_A</td>\n",
       "      <td>0.9418</td>\n",
       "      <td>0.7504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>s2</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs11538871&amp;COSV54528897</td>\n",
       "      <td>chr16:89911337-89911337</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.690796</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000539976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.471</td>\n",
       "      <td>1.630</td>\n",
       "      <td>2.038000e-04</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>chr16:89911337</td>\n",
       "      <td>T_C</td>\n",
       "      <td>0.7405</td>\n",
       "      <td>0.7227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>s2</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs74415461</td>\n",
       "      <td>chr16:89924435-89924435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.174964</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000625617</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.428</td>\n",
       "      <td>1.577</td>\n",
       "      <td>6.684000e-04</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>chr16:89924435</td>\n",
       "      <td>T_C</td>\n",
       "      <td>0.3897</td>\n",
       "      <td>0.1473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>s2</td>\n",
       "      <td>MC1R</td>\n",
       "      <td>rs74465604</td>\n",
       "      <td>chr16:89942655-89942655</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.985060</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000555576</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.369</td>\n",
       "      <td>1.511</td>\n",
       "      <td>1.035000e-03</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>chr16:89942655</td>\n",
       "      <td>C_T</td>\n",
       "      <td>1.0030</td>\n",
       "      <td>0.9895</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Signal Region       Existing_variation                   GRCh38 ref_alt  \\\n",
       "0     s1   MC1R                rs1108064  chr16:89827742-89827742     C_G   \n",
       "1     s1   MC1R       rs1805007&CM981238  chr16:89919709-89919709     T_C   \n",
       "2     s1   MC1R               rs74800773  chr16:89958562-89958562     G_C   \n",
       "3     s1   MC1R               rs45610233  chr16:89981987-89981987     T_C   \n",
       "4     s1   MC1R               rs73283859  chr16:89996112-89996112     C_G   \n",
       "5     s2   MC1R   rs1800282&COSV66881217  chr16:89816599-89816599     NaN   \n",
       "6     s2   MC1R  rs11538871&COSV54528897  chr16:89911337-89911337     NaN   \n",
       "7     s2   MC1R               rs74415461  chr16:89924435-89924435     NaN   \n",
       "8     s2   MC1R               rs74465604  chr16:89942655-89942655     NaN   \n",
       "\n",
       "   -logP_fisher                              Consequence  Protein_position  \\\n",
       "0      3.561616                    upstream_gene_variant               NaN   \n",
       "1     11.147276                    upstream_gene_variant               NaN   \n",
       "2      9.714218                  downstream_gene_variant               NaN   \n",
       "3      9.341321                  downstream_gene_variant               NaN   \n",
       "4     10.352324       non_coding_transcript_exon_variant               NaN   \n",
       "5      2.776504  missense_variant&NMD_transcript_variant               6.0   \n",
       "6      3.690796                    upstream_gene_variant               NaN   \n",
       "7      3.174964                           intron_variant               NaN   \n",
       "8      2.985060                  downstream_gene_variant               NaN   \n",
       "\n",
       "  Amino_acids          Feature CANONICAL  OR_fisher  OR_cond1      P_fisher  \\\n",
       "0         NaN  ENST00000378247       YES      1.253     1.108  2.744000e-04   \n",
       "1         NaN  ENST00000315491       YES      1.899     1.000  7.124000e-12   \n",
       "2         NaN  ENST00000564836       NaN      1.815     1.226  1.931000e-10   \n",
       "3         NaN  ENST00000423742       NaN      1.845     1.063  4.557000e-10   \n",
       "4         NaN  ENST00000437774       NaN      1.885     1.259  4.443000e-11   \n",
       "5         V/D  ENST00000563513       NaN      1.364     1.471  1.673000e-03   \n",
       "6         NaN  ENST00000539976       NaN      1.471     1.630  2.038000e-04   \n",
       "7         NaN  ENST00000625617       NaN      1.428     1.577  6.684000e-04   \n",
       "8         NaN  ENST00000555576       NaN      1.369     1.511  1.035000e-03   \n",
       "\n",
       "    P_cond1           38POS ref_alt_x  OR_cond2  P_cond2  \n",
       "0  0.108300  chr16:89827742       NaN       NaN      NaN  \n",
       "1  1.000000  chr16:89919709       NaN       NaN      NaN  \n",
       "2  0.250100  chr16:89958562       NaN       NaN      NaN  \n",
       "3  0.791600  chr16:89981987       NaN       NaN      NaN  \n",
       "4  0.328200  chr16:89996112       NaN       NaN      NaN  \n",
       "5  0.000107  chr16:89816599       T_A    0.9418   0.7504  \n",
       "6  0.000005  chr16:89911337       T_C    0.7405   0.7227  \n",
       "7  0.000020  chr16:89924435       T_C    0.3897   0.1473  \n",
       "8  0.000026  chr16:89942655       C_T    1.0030   0.9895  "
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
